stan/math/prim/prob/von_mises_cdf.hpp

```diff
@@ -0,0 +1,124 @@
+#ifndef STAN_MATH_PRIM_PROB_VON_MISES_CDF_HPP
+#define STAN_MATH_PRIM_PROB_VON_MISES_CDF_HPP
+
+#include <stan/math/prim/fun/constants.hpp>
+#include <stan/math/prim/prob/normal_cdf.hpp>
+#include <stan/math/prim/fun/modified_bessel_first_kind.hpp>
+#include <cmath>
+
+namespace stan {
+namespace math {
+
+namespace internal {
+
+template <typename T_x, typename T_k, typename T_p>
+return_type_t<T_x, T_k, T_p> von_mises_cdf_series(const T_x& x, const T_k& k,
+                                                  const T_p& p) {
+  using stan::math::cos;
+  using stan::math::sin;
+
+  double pi = stan::math::pi();
+  auto s = sin(x);
+  auto c = cos(x);
+  auto sn = sin(p * x);
+  auto cn = cos(p * x);
+
+  return_type_t<T_x, T_k, T_p> R = 0;
+  return_type_t<T_x, T_k, T_p> V = 0;
+
+  int n;
+  for (n = p - 1; n > 0; n--) {
+    auto sn_tmp = sn * c - cn * s;
+    cn = cn * c + sn * s;
+    sn = sn_tmp;
+    R = 1 / (2.0 * n / k + R);
+    V = R * (sn / n + V);
+  }
+  return 0.5 + x / (2 * pi) + V / pi;
+}
+
+template <typename T_x, typename T_k>
+return_type_t<T_x, T_k> von_mises_cdf_normalapprox(const T_x& x, const T_k& k) {
+  using std::exp;
+  using std::sqrt;
+  double pi = stan::math::pi();
+
+  auto b = sqrt(2 / pi) * exp(k) / modified_bessel_first_kind(0, k);
+  auto z = b * sin(x / 2);
+  double mu = 0;
+  double sigma = 1;
+
+  return normal_cdf(z, mu, sigma);
+}
+
+template <typename T_x, typename T_k>
+return_type_t<T_x, T_k> von_mises_cdf_centered(const T_x& x, const T_k& k) {
+  // if the scale is sufficiently small, approximate the cdf with a
+  // normal distribution, otherwise, use the expansion from scipy
```

Comment:
Usually we don't require docs for internal functions, but this seems to be doing something pretty specific and you have docs already under the signature so it may be good to include docs for this

Reply:
Yeah this have docs. This looks like the von-mises for mu = 0 and assuming a strict -pi to pi support right?

Just copy paste the docs from the other function and start with that. You don't need to repeat the equations in here.

Reply:
Yes, that's right. Will add the docs.